src/components/common/PageWithModuleTimeline/ModuleTimeline.tsx

```diff
@@ -46,6 +46,21 @@ export type ModuleTimelineProps = {
 };
 
 const ModuleTimeline = ({ cohortPhase, modules }: ModuleTimelineProps) => {
+  const theme = useTheme();
+  const now = new Date();
+  const futureModules = modules?.filter((module) => new Date(module.endDate) > now);
+
+  const soonestEndingModuleId = futureModules?.reduce((soonest, current) => {
+    const soonestEndDate = new Date(soonest.endDate);
+    const currentEndDate = new Date(current.endDate);
+    return currentEndDate < soonestEndDate ? current : soonest;
+  }, modules[0])?.id;
+
+  const warningLabelStyles = {
```

Comment:
Sorry my comments were confusing. Do we need these styles or does the `AltStepIcon` already handle it? 

I think the MuiStepLabel ones are only used if we use the default icon form the `StepLabel`

Reply:
We need those styles for the text label color, and the AltStepIcon styles are for the icon

Reply:
Looks good then! Thanks!!